### PR TITLE
COMMERCE-11152 SKU Virtual Settings

### PR DESCRIPTION
**Pull request by smottal**
https://issues.liferay.com/browse/COMMERCE-11152

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by smottal** [comment]
ci:test:sf

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/SkuVirtualSettingsDTOConverter.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+				termsOfUseContent = LanguageUtils.getLanguageIdMap(
+					cpDefinitionVirtualSetting.getTermsOfUseContentMap());
+				termsOfUseRequired =
+					cpDefinitionVirtualSetting.isTermsOfUseRequired();
+				url = cpDefinitionVirtualSetting.getUrl();
+				useSample = cpDefinitionVirtualSetting.isUseSample();
+
+				setActivationStatusInfo(
+					() -> {
+						String orderStatusLabel =
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/SkuVirtualSettingsDTOConverter.java:96]: I don't think this is the right name for this variable

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/SkuVirtualSettingsDTOConverter.java:96]: Now I fot it, it's right

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsResourceImpl.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+				contextAcceptLanguage.getPreferredLocale()));
+	}
+
+	@NestedField(parentClass = Sku.class, value = "skuVirtualSettings")
+	@Override
+	public SkuVirtualSettings getSkuIdSkuVirtualSettings(
+			@NestedFieldId(value = "id") Long id)
+		throws Exception {
+
+		CPInstance cpInstance = _cpInstanceService.fetchCPInstance(id);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsResourceImpl.java:70]: why don't you use the get method here?

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsResourceImpl.java (hunk 2 above)
Comment (smottal) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuVirtualSettingsResourceImpl.java]: Done

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/servlet/CommerceMediaServlet.java [hunk 3]
```diff
@@ -391,7 +394,26 @@ else if (CommerceMediaConstants.
...
-					httpServletRequest, httpServletResponse, true, pathArray);
+					httpServletRequest, httpServletResponse,
+					CPDefinition.class.getName(), true, pathArray);
+
+				return;
+			}
+			else if (CommerceMediaConstants.URL_SEPARATOR_VIRTUAL_SKU.contains(
+						pathArray[2])) {
+
+				_sendVirtualProductMediaBytes(
```
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/commerce/commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/servlet/CommerceMediaServlet.java]: Done

**Comment by smottal** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM, thank you!

**Comment by smottal** [comment]
ci:test:relevant

**Review by alessiorendina — APPROVED** [review]

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Pull request closed without merging**